aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -661,6 +661,13 @@ impl AptosVM {
                     }
                 }
             }
+            if let Some(metadata) = aptos_framework::get_module_metadata(m) {
+                if !aptos_framework::verify_metadata(&metadata) {
```

Comment:
Why not move the `Result` over there so this can be more generic (i.e. return different error messages from verify_metadata).

Reply:
absolutely